### PR TITLE
Dev env and Github actions

### PR DESCRIPTION
**Pull request by davidmh**
Since we're planning on modifying more than a few small things in this fork, we should:

- Enable local and CI environments using nix 
- Plug the tests into a Github action

### QA steps

Since you already have nix and direnv installed, all you need to install now is devenv as a standalone tool:

```shell
nix profile install nixpkgs#devenv
```

After that all you need to do is run `direnv allow` in the path for this project.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (davidmh) [inline .github/workflows/test.yml]: `macos-latest` is taking a while to run, we should drop it

**Comment by sgoel** [comment]
Qa steps worked for me!

**Review by sgoel — APPROVED** [review]

**Comment by sgoel** [comment]
@davidmh you mentioned in standup that this diff allows developers to not have to worry about the java dependencies on their system, but where is that defined? I don't see a JDK version being referenced anywhere

**Comment by davidmh** [comment]
Is set here: https://github.com/remix/OpenRailRouting/blob/166d3f9fa069c938cfe66a1b2fa79b83c4df10e4/devenv.nix#L4-L5

Since we're not specifying a jdk version, it goes with the default for the current release of devenv:

https://github.com/cachix/devenv/blob/main/src/modules/languages/java.nix#L17-L26

It installs java 21.0.2:

Screenshot 2024-12-18 at 13 03 19

Do we need a specific version?

**Pull request merged**